charts/oidc-proxy/templates/configmap.yaml

```diff
@@ -21,11 +21,27 @@ data:
 
     Header unset X-Frame-Options
     Header always set X-Frame-Options SAMEORIGIN
-    Header always set Access-Control-Allow-Origin *
     Header always set Access-Control-Allow-Headers DNT,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Range,Authorization,Accept,Referer,X-App-Id,Origin
     Header always set Access-Control-Allow-Methods GET,POST,DELETE,PUT,PATCH,OPTIONS,HEAD
     Header always set X-Content-Type-Options nosniff
 
+    # If CORS_REGEX environment variable is set, match the domains in the regex
+    # to set the Access-Control-Allow-Origin. Domains not in the regex will be
+    # denied.
+    <If "-n osenv('CORS_REGEX')">
+        SetEnvIf Origin "${CORS_REGEX}" ORIGIN_SUB_DOMAIN=$0
+    </If>
+    # If no CORS_REGEX environment variable is set, echo the Origin domain back
+    # in the Access-Control-Allow-Origin header.
+    <Else>
+        SetEnvIf Origin "(.+)" ORIGIN_SUB_DOMAIN=$0
+    </Else>
+    # CORS header will ONLY be set if the Origin header was sent. The browser
+    # does NOT set the Origin header on GETs: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Origin
+    # Due to this change Access-Control-Allow-Origin '*' is never set.
```

Comment:
Did you confirm that this would work with the WMS that this wouldn't break their workflow?

Reply:
Resolving per our internal conversation.